app/src/pages/SectionEdit.tsx

```diff
@@ -35,130 +54,155 @@ const SectionEdit = () => {
-
-    // Submit Handlers for function
-    const onExerciseAdd: SubmitHandler<Exercise> = data => addExercise(data);
-    const onSectionSave: SubmitHandler<Section> = data => saveSection(data);
-
-    // SubmitHandler: Add new exercise to section
-    const addExercise = async (data: Exercise) => {
-        const response = await ExerciseServices.addExercise(data, token, sid)
-        const addedExercise = response.data
-        sectionData.exercises.push(addedExercise)
-        setExercises(sectionData.exercises)
+  
+ /**
+     * Delete section and redirect to course edit page
+     * Uses window.location.href to redirect instead of navigate, as navigate doesn't update the page
+     * 
+     * @param sid The section id
+     * @param token The user token
+     */
+ const deleteSection = async () => {
+    const response = await SectionServices.deleteSection(sid, token);
+    const status = response.status
+
+    if (status >= 200 && status <= 299) {
+        window.location.href = `/courses/edit/${cid}`;
+        toast.success("Section deleted")
+    } else if (status >= 400 && status <= 599) {
+        toast.error(`(${status}, ${response.statusText}) while attempting to delete section`)
     }
+}
 
-    // SubmitHandler: Save section update
-    const saveSection = async (data: Section) => {
-        const toSave = { ...sectionData, ...data };
-        // TODO: Reinstate token
-        const response = await SectionServices.saveSection(toSave, sid/*, token*/);
+    
+    /**
+     * SubmitHandler: update section
+     * 
+     * @param data  The data to be updated
+    */
+   const onSubmit: SubmitHandler<Inputs> = (data) => {
+       const changes: SectionPartial = {
+           title: data.title,
+           description: data.description
+        }
+        
+        SectionServices.saveSection(changes, sid/*, token*/)
+        .then(res => toast.success('Seção atualizada'))
+        .catch(err => toast.error(err));
+    }
+    
+    /**
+     * Delete section and redirect to course edit page
+     * Uses window.location.href to redirect instead of navigate, as navigate doesn't update the page
+     * 
+     * @param sid The section id
+     * @param token The user token
+     *//*
+    const deleteSection = async () => {
+        const response = await SectionServices.deleteSection(sid, token);
         const status = response.status
 
         if (status >= 200 && status <= 299) {
-            toast.success("Section saved")
-            setSection(response.data);
+            window.location.href = `/courses/edit/${cid}`;
+            toast.success("Section deleted")
         } else if (status >= 400 && status <= 599) {
-            toast.error(`(${status}, ${response.statusText}) while attempting to save section`)
+            toast.error(`(${status}, ${response.statusText}) while attempting to delete section`)
         }
-    }
-
+    }*/
```

Comment:
dead code?

Reply:
Once again. Commented out since backend route is bad, but will be added in the next pull request.